drgn/helpers/linux/bpf.py

```diff
@@ -0,0 +1,76 @@
+# SPDX-License-Identifier: GPL-3.0+
+
+"""
+BPF
+---
+
+The ``drgn.helpers.linux.bpf`` module provides helpers for working with BPF
+interface in :linux:`include/linux/bpf.h`, :linux:`include/linux/bpf-cgroup.h`,
+etc.
+"""
+
+
+import enum
+
+from drgn.helpers.linux.list import list_for_each_entry
+
+
+__all__ = [
+    'get_bpf_attach_types',
+    'cgroup_bpf_prog_for_each',
+    'cgroup_bpf_prog_for_each_effective',
+]
+
+
+def get_bpf_attach_types(prog):
+    """
+    Get all BPF attach types as an :class:`enum.IntEnum` class.
+
+    :rtype: enum.IntEnum
+    """
+    enum_name = 'BpfAttachTypes'
+    try:
+        return prog.cache[enum_name]
+    except KeyError:
+        enumerators = [
+            (name, value)
+            for (name, value) in prog['__MAX_BPF_ATTACH_TYPE'].type_.enumerators
```

Comment:
> Since this type isn't anonymous, we can use `prog.type('enum bpf_attach_type')` instead of `prog['__MAX_BPF_ATTACH_TYPE'].type`. Also, we probably want to remove the redundant `BPF_` prefix?

Oh, `prog.type` -- that's definitely better. I tried to do it with `prog['enum bpf_attach_type']`, failed and decided to go with what worked. Sure, will switch to it.

As for `BPF_` prefix IMO it's better to keep it so that e.g. I can copy/paste it from UAPI to my drgn script. Also it simplifies auto-generating types for enums in general -- just use the name as is.

We did a different thing for TCP states enum, but that one is very common in many user-space utils and people got used to `LISTEN` more than `TCP_LISTEN` so there it makes sense.



> At this point I wonder if we want to factor this out into a utility function in `drgn/helpers/__init__.py`. Something like `get_enum(name, type, prefix)`? Feel free to suggest a better name.

Yep, I had same thoughts, but .. I was lazy :) I added the function in `drgn/helpers/__init__.py` and wen can bikesched on names in the next update.